backend/src/impl/benchmark_utils.py

```diff
@@ -331,4 +356,59 @@ def dataframe_to_table(
             system_names=list(scores.index),
             column_names=list(scores.columns),
             scores=[[scores[j][i] for j in scores.columns] for i in scores.index],
+            plot_y_values=[pt[1] for pt in plot_dict[view_name]],
+            plot_x_values=[pt[0] for pt in plot_dict[view_name]],
+        )
+
+    @staticmethod
+    def generate_plots(benchmark_id):
+        config = BenchmarkConfig.from_dict(
+            BenchmarkUtils.config_dict_from_id(benchmark_id)
         )
+        if config.type == "abstract":
+            return {}
+        plot_path = os.path.join(get_cache_dir(), benchmark_id + "_plot.json")
+        plot_file = open_cached_file(
+            benchmark_id + "_plot.json", datetime.timedelta(seconds=1)
+        )
+        if not plot_file:
+            sys_infos = BenchmarkUtils.load_sys_infos(config)
+            sorted_sys_infos = sorted(
+                sys_infos, key=lambda sys: sys["created_at"].date()
+            )
+            time = sorted_sys_infos[0]["created_at"]
+            current_time = datetime.datetime.now().date()
+            json_dict = {k.name: [] for k in config.views}
+            json_dict["Original"] = []
+            json_dict["times"] = []
+            while time.date() <= current_time:
```

Comment:
This will require a loop as long as the number of days from the beginning to the end. I would suggest doing something like this:
```
unique_dates = sorted(list(set([x["created_at"].date() for x in sys_infos])))
for date in unique_dates:
  ...
```
This will only process dates on which a system was submitted, probably saving a significant amount of time.

Reply:
That's a really great idea; I've changed it accordingly!